MediaBrowser.MediaEncoding/Subtitles/SubtitleEncoder.cs

```diff
@@ -542,6 +614,12 @@ private async Task ExtractAllExtractableSubtitlesInternal(
                 var outputCodec = IsCodecCopyable(subtitleStream.Codec) ? "copy" : "srt";
                 var streamIndex = EncodingHelper.FindIndex(mediaSource.MediaStreams, subtitleStream);
 
+                if (subtitleStream.Path.EndsWith(".mks", StringComparison.OrdinalIgnoreCase))
+                {
+                    _logger.LogDebug("Subtitle {Index} for file {InputPath} is part in an MKS file. Skipping", inputPath, subtitleStream.Index);
+                    continue;
+                }
+
```

Comment:
Move it to the beginning of the loop to skip redundant `outputPath`, `outputCodec` and `streamIndex`.

Reply:
Done.